doc/source/release/2.0.0-notes.rst

```diff
@@ -4,6 +4,93 @@
 NumPy 2.0.0 Release Notes
 =========================
 
+NumPy 2.0.0 is the first major release since 2006. It is the result of X months
+of development since the last feature release by Y contributors, and contains a
+large amount of exciting new features as well as a large amount of changes to
+both the Python and C APIs.
```

Comment:
I am wondering if we can expand this section in some free-form to include:

Beyond general improvements and new features the major changes for Python users are Python API changes which for the most part are removals and cleanups but may require code adaptations (NEP 56) and the changes in promotion (NEP 50) which fix many user surprises but can lead to different precision results.

I.e. these are the most relevant (to end users) changes which couldn't happen in a 1.x release.  NEP 50 feels a bit lost below considering that it has quite a bit of potential for breaking things.
(The migration guide makes that clear I think, so I guess if you don't like listing anything here, you could also just stress that as a major release 2.0 had some changes which could affect users more, please see there there...)

Reply:
good idea, added something along these lines.